sessions/workflow_dacx.go

```diff
@@ -0,0 +1,90 @@
+	err := processFile(ctx, param)
+	return err
+}
+
+func processFile(ctx workflow.Context, param FileProcessingWFParam) error {
+	// Set the required timeouts for the Worker Session
+	sessionOptions := &workflow.SessionOptions{
+		CreationTimeout:  time.Minute,
+		ExecutionTimeout: time.Minute,
+	}
+	// Create a Session with the Worker so that all Activities execute with the same Worker.
+	sessionCtx, err := workflow.CreateSession(ctx, sessionOptions)
+	if err != nil {
+		return err
+	}
+	defer workflow.CompleteSession(sessionCtx)
+
+	var a *FileActivities
+	var downloadResult FileActivityResult
+	err = workflow.ExecuteActivity(sessionCtx, a.DownloadFile, param).Get(sessionCtx, &downloadResult)
+	if err != nil {
+		return err
+	}
+	processParam := FileActivityParam(downloadResult)
+	var processResult FileActivityResult
+	err = workflow.ExecuteActivity(sessionCtx, a.ProcessFile, processParam).Get(sessionCtx, &processResult)
+	if err != nil {
+		return err
+	}
+	uploadParam := FileActivityParam(processResult)
+	err = workflow.ExecuteActivity(sessionCtx, a.UploadFile, uploadParam).Get(sessionCtx, nil)
+	return err
+}
+
+/*
+When the `CreateSession` API is called, the Task Queue name that is specified in the `ActivityOptions` is used (or in the `StartWorkflowOptions` if the Task Queue name is not specified in `ActivityOptions`), and a Session is created with one of the Workers polling that Task Queue.
+
+The Session Context is cancelled if the Worker executing this Session dies or `CompleteSession()` is called.
+When using the returned Session Context to spawn Activity Executions, a `workflow.ErrSessionFailed` error may be returned if the Session framework detects that the Worker executing this Session has died.
+The failure of Activity Executions won't affect the state of the Session, so you still need to handle the errors returned from your Activities and call `CompleteSession()` if necessary.
+
+`CreateSession()` will return an error if the context passed in already contains an open Session.
+If all the Workers are currently busy and unable to handle a new Session, the framework will keep retrying until the `CreationTimeout` you specified in `SessionOptions` has passed before returning an error (check the **Concurrent Session Limitation** section for more details).
+
+`CompleteSession()` releases the resources reserved on the Worker, so it's important to call it as soon as you no longer need the Session.
+It will cancel the session context and therefore all the Activity Executions using that Session Context.
+It is safe to call `CompleteSession()` on a failed Session, meaning that you can call it from a `defer` function after the Session is successfully created.
+
+If the Worker goes down between Activities, any scheduled Activities meant for the Session Worker are canceled.
+If not, you will get a `workflow.ErrSessionFailed` error when the next call of `workflow.ExecuteActivity()` is made from that Workflow.
```

Comment:
```suggestion
If not, you get a `workflow.ErrSessionFailed` error when the next call of `workflow.ExecuteActivity()` is made from that Workflow.
```
To what situation does "If not" refer? A failure of scheduled Activities to be canceled?